File: Dice.java
// Dice.java
import java.util.*;
import java.io.*;
import java.nio.file.*;
import java.time.Instant;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public class Dice {
    private final int sides;
    private final Random random;

    public Dice(int sides, Long seed) {
        this.sides = sides;
        this.random = (seed != null) ? new Random(seed) : new Random();
    }

    public int[] roll(int count) {
        int[] results = new int[count];
        for (int i = 0; i < count; i++) {
            results[i] = random.nextInt(sides) + 1;
        }
        return results;
    }

    public static void main(String[] args) throws Exception {
        // Простой парсинг аргументов (без библиотек)
        Map<String, String> params = new HashMap<>();
        for (int i = 0; i < args.length; i++) {
            if (args[i].startsWith("--")) {
                String key = args[i].substring(2);
                if (i + 1 < args.length && !args[i+1].startsWith("--")) {
                    params.put(key, args[++i]);
                } else {
                    params.put(key, "");
                }
            }
        }

        int roll = Integer.parseInt(params.getOrDefault("roll", "1"));
        int count = Integer.parseInt(params.getOrDefault("count", "1"));
        Long seed = params.containsKey("seed") ? Long.parseLong(params.get("seed")) : null;
        boolean stats = params.containsKey("stats");
        String exportJson = params.get("export-json");
        String exportCsv = params.get("export-csv");

        Dice dice = new Dice(6, seed);
        int[][] results = new int[count][];
        Map<Integer, Integer> statsMap = new HashMap<>();
        int totalSum = 0;

        System.out.println("\u001B[36mБросаем " + roll + " костей, " + count + " раз(а)\u001B[0m");
        for (int i = 0; i < count; i++) {
            int[] vals = dice.roll(roll);
            results[i] = vals;
            int sum = 0;
            for (int v : vals) {
                sum += v;
                statsMap.put(v, statsMap.getOrDefault(v, 0) + 1);
            }
            totalSum += sum;
            System.out.printf("\u001B[32mБросок %d: [%s] сумма = %d\u001B[0m%n", i+1, Arrays.toString(vals), sum);
        }

        if (count > 1) {
            double avg = (double) totalSum / count;
            System.out.printf("\u001B[33mОбщая сумма всех бросков: %d, среднее: %.2f\u001B[0m%n", totalSum, avg);
        }

        if (stats) {
            System.out.println("\u001B[35m\nСтатистика выпадений:\u001B[0m");
            for (int side = 1; side <= 6; side++) {
                System.out.printf("  %d: %d%n", side, statsMap.getOrDefault(side, 0));
            }
        }

        Gson gson = new GsonBuilder().setPrettyPrinting().create();

        if (exportJson != null) {
            Map<String, Object> data = new LinkedHashMap<>();
            data.put("timestamp", Instant.now().toString());
            data.put("roll_count", roll);
            data.put("num_rolls", count);
            data.put("results", results);
            data.put("statistics", statsMap);
            data.put("total_sum", totalSum);
            data.put("average", (double) totalSum / count);
            String json = gson.toJson(data);
            Files.write(Paths.get(exportJson), json.getBytes());
            System.out.println("\u001B[34mРезультаты экспортированы в " + exportJson + "\u001B[0m");
        }

        if (exportCsv != null) {
            try (PrintWriter pw = new PrintWriter(new FileWriter(exportCsv))) {
                pw.println("roll_number,values,sum");
                for (int i = 0; i < count; i++) {
                    int[] vals = results[i];
                    int sum = 0;
                    StringBuilder sb = new StringBuilder();
                    for (int j = 0; j < vals.length; j++) {
                        if (j > 0) sb.append(",");
                        sb.append(vals[j]);
                        sum += vals[j];
                    }
                    pw.printf("%d,\"%s\",%d%n", i+1, sb.toString(), sum);
                }
            }
            System.out.println("\u001B[34mРезультаты экспортированы в " + exportCsv + "\u001B[0m");
        }
    }
}

File: Dice.kt
// Dice.kt
import com.beust.jcommander.JCommander
import com.beust.jcommander.Parameter
import com.google.gson.GsonBuilder
import java.io.File
import java.time.Instant
import kotlin.random.Random

class Dice(private val sides: Int, seed: Long? = null) {
    private val random = if (seed != null) Random(seed) else Random(System.currentTimeMillis())

    fun roll(count: Int): List<Int> {
        return (1..count).map { random.nextInt(1, sides + 1) }
    }
}

class Options {
    @Parameter(names = ["--roll"], description = "Количество костей в броске")
    var roll: Int = 1

    @Parameter(names = ["--count"], description = "Количество бросков")
    var count: Int = 1

    @Parameter(names = ["--seed"], description = "Seed для генератора")
    var seed: Long? = null

    @Parameter(names = ["--stats"], description = "Показать статистику")
    var stats: Boolean = false

    @Parameter(names = ["--export-json"], description = "Экспорт в JSON")
    var exportJson: String? = null

    @Parameter(names = ["--export-csv"], description = "Экспорт в CSV")
    var exportCsv: String? = null
}

fun main(args: Array<String>) {
    val options = Options()
    JCommander.newBuilder().addObject(options).build().parse(*args)

    val dice = Dice(6, options.seed)
    val results = mutableListOf<List<Int>>()
    val stats = mutableMapOf<Int, Int>()
    var totalSum = 0

    println("\u001B[36mБросаем ${options.roll} костей, ${options.count} раз(а)\u001B[0m")
    for (i in 0 until options.count) {
        val vals = dice.roll(options.roll)
        results.add(vals)
        val sum = vals.sum()
        totalSum += sum
        for (v in vals) {
            stats[v] = stats.getOrDefault(v, 0) + 1
        }
        println("\u001B[32mБросок ${i+1}: [${vals.joinToString(", ")}] сумма = $sum\u001B[0m")
    }

    if (options.count > 1) {
        val avg = totalSum.toDouble() / options.count
        println("\u001B[33mОбщая сумма всех бросков: $totalSum, среднее: ${"%.2f".format(avg)}\u001B[0m")
    }

    if (options.stats) {
        println("\u001B[35m\nСтатистика выпадений:\u001B[0m")
        for (side in 1..6) {
            println("  $side: ${stats[side] ?: 0}")
        }
    }

    options.exportJson?.let { file ->
        val data = mapOf(
            "timestamp" to Instant.now().toString(),
            "roll_count" to options.roll,
            "num_rolls" to options.count,
            "results" to results,
            "statistics" to stats,
            "total_sum" to totalSum,
            "average" to totalSum.toDouble() / options.count
        )
        val gson = GsonBuilder().setPrettyPrinting().create()
        File(file).writeText(gson.toJson(data))
        println("\u001B[34mРезультаты экспортированы в $file\u001B[0m")
    }

    options.exportCsv?.let { file ->
        File(file).printWriter().use { pw ->
            pw.println("roll_number,values,sum")
            results.forEachIndexed { idx, vals ->
                val sum = vals.sum()
                pw.println("${idx+1},\"${vals.joinToString(",")}\",$sum")
            }
        }
        println("\u001B[34mРезультаты экспортированы в $file\u001B[0m")
    }
}
